api/python/quilt3/hooks.py

```diff
@@ -0,0 +1,60 @@
+import typing as T
+
+import boto3
+
+
+class BuildClientBase(T.Protocol):
+    def __call__(self, session: boto3.Session, client_kwargs: dict[str, T.Any], **kwargs): ...
+
+
+class BuildClientHook(T.Protocol):
+    def __call__(
+        self, build_client_base: BuildClientBase, session: boto3.Session, client_kwargs: dict[str, T.Any], **kwargs
+    ): ...
+
+
+_build_client_hook = None
+
+
+def get_build_s3_client_hook() -> T.Optional[BuildClientHook]:
+    """
+    Return build S3 client hook.
+    """
+
+    return _build_client_hook
+
+
+def set_build_s3_client_hook(hook: T.Optional[BuildClientHook]) -> T.Optional[BuildClientHook]:
+    """
+    Set build S3 client hook.
+
+    Example for overriding `ServerSideEncryption` parameter for certain S3 operations:
+
+    ```python
+    def event_handler(params, **kwargs):
+        # Be mindful with parameters you set here.
+        # Specifically it's not recommended to override/delete already set parameters
+        # because that can break quilt3 logic.
+        params.setdefault("ServerSideEncryption", "AES256")
+
+    def hook(build_client_base, session, client_kwargs, **kwargs):
+        client = build_client_base(session, client_kwargs, **kwargs)
+        for op in (
+            "CreateMultipartUpload",
+            "CopyObject",
+            "PutObject",
+        ):
+            client.meta.events.register(f"before-parameter-build.s3.{op}", event_handler)
```

Comment:
I assume `before-parameter-build.s3.*` are the standard boto3 events?   
Is that obvious to everyone, or would it help to mention the documentation?
https://boto3.amazonaws.com/v1/documentation/api/latest/guide/events.html

Reply:
adding the link to the boto docs sounds like a good idea